packages/astro/src/content/vite-plugin-content-imports.ts

```diff
@@ -55,14 +68,46 @@ export function astroContentImportPlugin({
 }): Plugin[] {
 	const contentPaths = getContentPaths(settings.config, fs);
 	const contentEntryExts = getContentEntryExts(settings);
+	const dataEntryExts = getDataEntryExts(settings);
 
 	const contentEntryConfigByExt = getContentEntryConfigByExtMap(settings);
 
+	const dataEntryExtToParser: Map<string, DataEntryType['getEntryInfo']> = new Map();
+	for (const entryType of settings.dataEntryTypes) {
+		for (const ext of entryType.extensions) {
+			dataEntryExtToParser.set(ext, entryType.getEntryInfo);
+		}
+	}
+
 	const plugins: Plugin[] = [
 		{
 			name: 'astro:content-imports',
 			async transform(_, viteId) {
-				if (isContentFlagImport(viteId)) {
+				if (hasContentFlag(viteId, DATA_FLAG)) {
+					const fileId = viteId.split('?')[0];
```

Comment:
Same here, we should leave a comment saying why it's safe to query the index 0.

Reply:
JS returns `undefined` even when an index is out-of-bounds, so I'm not sure it's worth the comment?

Reply:
Looks great! Thank you!